src/SimulationRunner.cc

```diff
@@ -705,6 +705,15 @@ void SimulationRunner::Step(const UpdateInfo &_info)
   // Update all the systems.
   this->UpdateSystems();
 
+  if (!this->Paused() &&
+       this->requestedRunToSimTime >
+       std::chrono::steady_clock::duration::zero() &&
+       this->currentInfo.simTime >= this->requestedRunToSimTime)
+  {
+    this->SetPaused(true);
```

Comment:
Should we also execute `--this->pendingSimIterations;` as the next block? Alternatively, we could move this condition within the next block and combine it with:

```
if (this->pendingSimIterations <= 0)
```

as the result is the same (to pause the simulation).

Reply:
No, we don't need to decrement pending sim iterations here since that is a separate feature.